### PR TITLE
InitTrigger enum is used in config to check genesis state

### PR DESCRIPTION
**Pull request by hendrikebbers**
InitTrigger enum is used in config to check genesis state

### CODE REVIEW

**Review by timo0 — DISMISSED** [review]

**Comment by rbair23** [comment]
I'm not sure if I like this. We couple the config to InitTrigger, but we only use a single state of the InitTrigger, which could be just a boolean.

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/config/ConfigProviderImplTest.java [hunk 1]
```diff
@@ -65,10 +66,28 @@ void testApplicationPropertiesLoaded() {
...
+    void testCorrectInitTriggerUsage(final InitTrigger initTrigger) {
+        // given
+        final var configProvider = new ConfigProviderImpl(initTrigger);
+
+        // when
+        final var configuration = configProvider.getConfiguration();
+        final String value = configuration.getValue("bar.test");
+
+        // then
+        if (Objects.equals(initTrigger, InitTrigger.GENESIS)) {
```
Comment (jsync-swirlds) [inline hedera-node/hedera-app/src/test/java/com/hedera/node/app/config/ConfigProviderImplTest.java:80]: This can use `==` here as well.

**Comment by hendrikebbers** [comment]
@rbair23 the `InitTrigger` is managed by Dagger as a singleton. If we want to have a boolean as a constructor param we need to create a custom `@Qualifier` annotation to define the boolean:

`public ConfigProviderImpl(@NonNull @GenesisQualifier final Boolean genesis)`

That was my first approach but it makes the API even harder to read from my point of view. I assume we need to decide if we want to remove the dependency to `InitTrigger` or add an annotation only for this specific case (+ adding code that provides the boolean value based on the `InitTrigger`).

If anyone sees a better way to solve this issue I'm super happy to hear about it :)

**Comment by rbair23** [comment]
> @rbair23 the `InitTrigger` is managed by Dagger as a singleton. If we want to have a boolean as a constructor param we need to create a custom `@Qualifier` annotation to define the boolean:
> 
> `public ConfigProviderImpl(@NonNull @GenesisQualifier final Boolean genesis)`
> 
> That was my first approach but it makes the API even harder to read from my point of view. I assume we need to decide if we want to remove the dependency to `InitTrigger` or add an annotation only for this specific case (+ adding code that provides the boolean value based on the `InitTrigger`).
> 
> If anyone sees a better way to solve this issue I'm super happy to hear about it :)

I have an upcoming PR where the config provider will be created directly by Hedera.java and supplied as an argument to the dagger object rather than created by dagger. This is because I have to control creation of the object in terms of when it is created and whether it is genesis or not.

**Review by timo0 — APPROVED** [review]

**Pull request closed without merging**